ros/src/common/libs/state_machine_lib/src/state_context.cpp

```diff
@@ -23,358 +23,335 @@
+  std::string text;
+
+  std::shared_ptr<State> state = root_state_;
+  do
   {
-    if (_kind == BEHAVIOR_STATE)
-    {
-      uint64_t _current_state = HolderMap[_kind];
-      std::string ret = "";
-      for (uint64_t mask = STATE_SUB_END; _current_state != 0 && mask != 0; mask >>= 1)
-      {
-        if (mask & _current_state)
-        {
-          ret += "\n" + getStateName(mask);
-          _current_state &= ~mask;
-        }
-      }
-      return ret;
-    }
-    else
+    for (const auto& keyval : state->getTransitionMap())
     {
-      return getStateName(HolderMap[_kind]);
+      text += keyval.first + ":" + state_map_[keyval.second]->getStateName() + ",";
     }
-  }
-  return std::string("");
+    state = state->getChild();
+  } while (state != nullptr);
+
+  return text;
 }
 
-BaseState *StateContext::getCurrentMainState(void)
+std::string StateContext::getStateText(void)
 {
-  return getStateObject(HolderMap[MAIN_STATE]);
+  std::string text;
+
+  std::shared_ptr<State> state = root_state_;
+  do
+  {
+    text += state->getStateName() + "\n";
+    state = state->getChild();
+  } while (state != nullptr);
+
+  return text;
 }
 
-bool StateContext::disableCurrentState(uint64_t _state_num)
+void StateContext::setTransitionMap(const YAML::Node& node, const std::shared_ptr<State>& _state)
 {
-  if (isMainState(getStateObject(_state_num)))
-  {
-    return false;
-  }
-  if (isCurrentState(_state_num))
-  {
-    HolderMap[getStateKind(_state_num)] &= ~_state_num;
-    getStateObject(_state_num)->outState();
-    return true;
-  }
-  else
+  for (unsigned int j = 0; j < node.size(); j++)
   {
-    return false;
+    _state->addTransition(node[j]["Key"].as<std::string>(), getStateIDbyName(node[j]["Target"].as<std::string>()));
   }
 }
 
-bool StateContext::isCurrentState(uint64_t _state_num)
+void StateContext::showStateName()
 {
-  if (_state_num)
-    return HolderMap[getStateKind(_state_num)] & _state_num;
-  else
-    return false;
+  std::shared_ptr<State> state = root_state_;
+  do
+  {
+    state = state->getChild();
+  } while (state != nullptr);
 }
 
-bool StateContext::isState(BaseState *base, uint64_t _state_num)
+std::shared_ptr<State> StateContext::getStatePtr(const YAML::Node& node)
 {
-  return base ? base->getStateNum() & _state_num ? true : false : false;
+  return getStatePtr(node["StateName"].as<std::string>());
 }
 
-bool StateContext::handleIntersection(bool _hasIntersection, double _angle)
+std::shared_ptr<State> StateContext::getStatePtr(const std::string& _state_name)
 {
-  /* deprecated */
-  return false;
+  if (_state_name == "~")
+    return nullptr;
+  else
+    return getStatePtr(getStateIDbyName(_state_name));
 }
 
-std::string StateContext::createStateMessageText(void)
+std::shared_ptr<State> StateContext::getStatePtr(const uint64_t& _state_id)
 {
-  std::string ret;
+  return state_map_[_state_id];
+}
 
-  for (auto &p : HolderMap)
+void StateContext::parseChildTransitionMap(const YAML::Node& node)
```

Comment:
This private member function seems not to be used anywhere, should be removed if not needed